packages/core/src/domain/session/sessionStoreOperations.ts

```diff
@@ -20,15 +23,37 @@ export function processSessionStoreOperations(
   numberOfRetries = 0
 ) {
   const { isLockEnabled, persistSession, expireSession } = sessionStoreStrategy
-  const persistWithLock = (session: SessionState) => persistSession({ ...session, lock: currentLock })
+  const persistWithLock = (session: SessionState) =>
+    persistSession({ ...session, lock: currentLock, lockTimestamp: String(dateNow()) })
   const retrieveStore = () => {
     const session = sessionStoreStrategy.retrieveSession()
-    const lock = session.lock
+    let lock = session.lock
+
+    if (lock && session.lockTimestamp) {
```

Comment:
I think we should also delete the lock if `(lock && ! session.lockTimestamp)` so we clear the cookies that are already in a locked state before this change is deployed